singlepost.jsx

```diff
@@ -1,73 +1,304 @@
+// SinglePost.js
 import React, { useEffect, useState } from 'react';
 
-const PostContainer = () => {
-  const [isLiked, setIsLiked] = useState(false);
-  const commentsData = [
-    { username: "Vizthewiz", text: "Great photo!" },
-    { username: "Samsamzebra", text: "Love it!" },
-    { username: "Miilovescats", text: "Awesome!" },
-    // Add more comments as needed
-  ];
-
-  useEffect(() => {
-    const likeButton = document.querySelector('.like-button');
-    const heartIcon = likeButton.querySelector('i');
-
-    likeButton.addEventListener('click', () => {
-      // Toggle heart icon
-      heartIcon.classList.toggle('far', !isLiked);
-      heartIcon.classList.toggle('fas', isLiked);
-
-      // Toggle isLiked
-      setIsLiked(!isLiked);
-    });
-  }, [isLiked]);
-
-  return (
-    <div className="post-container">
-      <div className="post-header">
-        <div className="user-info">
-          <img src="bird pixel.jpg" alt="User Avatar" />
-          <p className="username">FirstUser101</p>
+const SinglePost = () => {
+    const [isLiked, setIsLiked] = useState(false);
+    const [commentsData, setCommentsData] = useState([
+        { username: "Vizthewiz", text: "Great photo!" },
+        { username: "Samsamzebra", text: "Love it!" },
+        { username: "Miilovescats", text: "Awesome!" },
+    ]);
+    const [teaserComments] = useState([
+        { username: "FakeUser1", text: "Nice photo!" },
+        { username: "FakeUser2", text: "Fire photo!" },
+        { username: "FakeUser3", text: "Stunning shot!" },
+    ]);
+
+    useEffect(() => {
+        const likeButton = document.querySelector('.like-button');
+        const shareButton = document.querySelector('.interaction-buttons.share');
+        const likesCountElement = document.querySelector('.likes-count');
+        const sharesCountElement = document.querySelector('.shares-count');
```

Comment:
Awesome work so far! Just a couple of suggestions regarding React best practices. 

1. Same suggestions as everyone else in making sure nothing is hardcoded; Should be pulling the info from the backend (dm in the frontend chat if help is needed, seems like it might get complicated). 
2. It looks like all the functions and icons are wrapped in a huge useEffect. This will lead to a lot of performance issues later on as a useEffect often reloads the component multiple times depending on any change made to any of these buttons. If you wrap these icons in a <button> tag, the button tag has special operator like onClick or onScroll etc. that you can attach the functions to so that they run as intended without the need of wrapping everything in a useEffect. My suggestion would be to wrap those icons in the button tag and move all of this logic outside of the useEffect. 

Here's some resources: 
https://react.dev/learn/responding-to-events 
https://react.dev/reference/react/useEffect